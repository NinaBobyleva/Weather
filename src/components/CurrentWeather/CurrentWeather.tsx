export function CurrentWeather({cityName}: {cityName: string | undefined}) {

    return (
        <div className="border-4 h-[400px] w-[600px] rounded-3xl">
            <div>
                <h3>Погода в городе {cityName}</h3>
            </div>
            <div></div>
        </div>
    );
}