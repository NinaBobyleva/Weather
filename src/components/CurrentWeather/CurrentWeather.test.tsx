import * as renderer from "react-test-renderer";
import { CurrentWeather } from "./CurrentWeather";

describe("Компонент CurrentWeather", () => {
  it("Успешный рендер компонента CurrentWeather", () => {
    const component = renderer
      .create(
        <CurrentWeather
          cityName="Москва"
          weatherCurrent={{
            humidity: 97,
            temp: 27,
            dt: 1733302168,
            wind_speed: 3.27,
            weather: [],
          }}
          timezoneOffset={10800}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
