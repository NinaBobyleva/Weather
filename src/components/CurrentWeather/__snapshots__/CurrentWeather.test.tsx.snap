// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент CurrentWeather Успешный рендер компонента CurrentWeather 1`] = `
<div
  className="py-[32px] px-[4.5px] md:px-[10px] w-[320px] h-[270px] md:h-[400px] md:w-[500px] rounded-3xl bg-blue-100"
>
  <div
    className="flex flex-col items-center"
  >
    <h3
      className="text-[22px] md:text-[36px] md:pb-1"
    >
      Погода 
      Москва
    </h3>
    <span
      className="text-[16px] md:text-[32px] text-slate-400 text-left"
    >
      Сегодня 
      ср
    </span>
  </div>
  <div
    className="flex flex-nowrap justify-between items-center gap-[5px] md:gap-[16px] md:py-2"
  >
    <div
      className="flex w-[97px] md:w-[130px]"
    >
      <p
        className="text-[42px] md:text-[60px] pl-8"
      >
        27
      </p>
      <img
        alt="Градусы Цельсия"
        className="w-[12px] h-[12px] md:w-[20px] md:h-[20px] relative top-[15px] md:top-[20px]"
        src="/public/img/gradus.png"
      />
    </div>
    <img
      alt=""
      className=" w-[97px] md:w-[130px]"
      src="https://openweathermap.org/img/wn/@2x.png"
    />
    <p
      className="text-[14px] w-[97px] text-center md:text-[26px] md:w-[150px]"
    />
  </div>
  <div
    className="flex flex-nowrap justify-around"
  >
    <div
      className="flex flex-row items-center gap-1"
    >
      <img
        alt="Скорость ветра"
        className="w-[25px] md:w-[30px] h-[25px] md:h-[30px]"
        src="./img/icons8-ветер-40.png"
      />
      <p
        className="text-[28px] md:text-[32px]"
      >
        3
         
        м/с
      </p>
    </div>
    <div
      className="flex flex-row items-center gap-1"
    >
      <img
        alt="Влажность"
        className="w-[25px] md:w-[30px] h-[25px] md:h-[30px]"
        src="./img/icons8-влажный-40.png"
      />
      <p
        className="text-[28px] md:text-[32px]"
      >
        97
        %
      </p>
    </div>
  </div>
</div>
`;
